development-process.md

```diff
@@ -0,0 +1,144 @@
+
+Eventually you'll want to push this new branch to origin to make it available to other developers.
+You can do this before you open a Pull Request (see below) or right after you create the branch:
+
+    git push -u origin NEW-BRANCH_NAME
+
+#### &#10146; Hack away!
+
+Make changes, test (automated and/or manual), commit, repeat. The repo README should have details
+for running automated tests.
+
+
+### Step 2) Code Review
+
+_When you've completed the given issue, the code is passing tests, and you feel like it's ready
+to ship, it's time to start code review. Make sure you read through the
+[Code Review guide](https://github.com/mitodl/handbook/blob/master/code_review.md) to get a sense
+about why we do code reviews and how to approach them, both as a reviewer and reviewee._
+
+#### &#10146; Open a Pull Request ("PR")
+
+This can be kicked off from several places in Github. The easiest one to remember is the "New
+Pull Request" button in your repo's "Pull Requests" tab
+([example](https://github.com/mitodl/micromasters/pulls)). Your **base** branch should be
+**master** and your **compare** branch should be your issue branch.
+
+When you first open the PR, the input should be pre-populated with our
+[PR template](https://github.com/mitodl/handbook/blob/master/pr-template.md). Fill in every section
+that is relevant to your PR. Under the **"relevant tickets"** section, be sure to indicate the
+issue that you're fixing. Github will automatically link to your issue if you type '#' followed
+by your issue number (and it will help you with an autocomplete menu). **If your PR is intended to
+fix/close an issue (typically true), fill in this section with "fixes #123" or "closes #123".**
+When your PR is eventually merged, this text is parsed, and the issue will be automatically closed
+if "fixes #X"/"closes #X" is found.
+
+Before (or after) you submit the PR, add the "Needs Review" label to it.
+
+**TL;DR**
+1. Click "New Pull Request"
+1. Fill in (or confirm) the appropriate **base** and **compare** branches
+1. Fill in all the relevant sections in the PR template (especially "relevant tickets", which should
+   typically read "fixes #<issue_number>")
+1. Add the "Needs Review" label
+1. Submit and wait for comments/approval
+
+#### &#10146; Respond to comments until you get approval
+
+As comments come in and changes are requested, you can make any necessary changes and
+
+Reviewers should change the PR's label to "Waiting on author" when they are finished making comments
+and requesting changes. Similarly, when you are finished making changes based on that feedback and
+responding to comments, you should change the label back to "Needs Review".
+
+Right now, we consider a PR to be "approved" if the assigned reviewer on the PR has given you a +1
+(thumbs up). Use your own discretion about seeking out further approvals. That might be appropriate
+if there is someone very familiar with the code you're working on, but isn't the assigned reviewer.
```

Comment:
I would mention that if there is a disagreement, the final decision should be taken by the developer and not the reviewer (unless there are some substantial disagreements, then other people should probably be involved, but this should ideally never happen)